src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -300,19 +300,19 @@ Dictionary<string, ITaskItem> MaybeGetArchAssemblies (Dictionary<AndroidTargetAr
 
 		void GenerateAdditionalProviderSources (NativeCodeGenState codeGenState, IList<string> additionalProviders)
 		{
-			if (androidRuntime != Xamarin.Android.Tasks.AndroidRuntime.MonoVM) {
+			bool isMonoVm = androidRuntime == Xamarin.Android.Tasks.AndroidRuntime.MonoVM;
```

Comment:
I'm uncertain about this structure, as it's implicitly stating that CoreCLR & NativeAOT will be `JavaInterop1`-style JCWs, while MonoVM is `XAJavaInterop1`-style JCWs.  CoreCLR should probably use XAJavaInterop1, and while CoreCLR is in-progress (#9572), I don't think we want to introduce potential future "confusion".

Instead, I think we should consider either (1) adding a `CodeGenerationTarget` property to `<GenerateJavaStubs/>`, or (2):

```csharp
CodeGenerationTarget jcwType = androidRuntime switch {
  Xamarin.Android.Tasks.AndroidRuntime.NativeAOT => CodeGenerationTarget.JavaInterop1,
  _ => CodeGenerationTarget.XAJavaInterop1
};
```

then compare against the "normalized" `jcwType` value.